Source/NSStoryboard.m

```diff
@@ -0,0 +1,967 @@
+
+- (void) setStoryboardName: (NSString *)name
+{
+  ASSIGNCOPY(_storyboardName, name);
+}
+
+- (id) copyWithZone: (NSZone *)z
+{
+  NSControllerPlaceholder *c = [[NSControllerPlaceholder allocWithZone: z] init];
+  [c setStoryboardName: _storyboardName];
+  return c;
+}
+
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super init];
+  if ([coder allowsKeyedCoding])
+    {
+      if ([coder containsValueForKey: @"NSStoryboardName"])
+        {
+          [self setStoryboardName: [coder decodeObjectForKey: @"NSStoryboardName"]];
+        }
+    }
+  return self;
+}
+
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  // this is never encoded directly...
+}
+
+- (id) instantiate
+{
+  NSStoryboard *sb = [NSStoryboard storyboardWithName: _storyboardName
+                                               bundle: [NSBundle mainBundle]];
+  return [sb instantiateInitialController];
+}
+
+@end
+
+@implementation NSStoryboard
+
+- (NSXMLElement *) _createCustomObjectWithId: (NSString *)ident
+                                   userLabel: (NSString *)userLabel
+                                 customClass: (NSString *)className
+{
+  NSXMLElement *customObject =
+    [[NSXMLElement alloc] initWithName: @"customObject"];
+  NSXMLNode *idValue =
+    [NSXMLNode attributeWithName: @"id"
+                     stringValue: ident];
+  NSXMLNode *usrLabel =
+    [NSXMLNode attributeWithName: @"userLabel"
+                     stringValue: userLabel];
+  NSXMLNode *customCls =
+    [NSXMLNode attributeWithName: @"customClass"
+                     stringValue: className];
+  
+  [customObject addAttribute: idValue];
+  [customObject addAttribute: usrLabel];
+  [customObject addAttribute: customCls];
+
+  AUTORELEASE(customObject);
+  
+  return customObject;
+}
+
+- (void) _processStoryboard: (NSXMLDocument *)storyboardXml
+{
+  NSArray *docNodes = [storyboardXml nodesForXPath: @"document" error: NULL];
+
+  if ([docNodes count] > 0)
+    {
+      NSXMLElement *docNode = [docNodes objectAtIndex: 0];
+      NSArray *array = [docNode nodesForXPath: @"//scene" error: NULL];
+      NSString *customClassString = nil;
+      
+      // Set initial view controller...
+      ASSIGN(_initialViewControllerId, [[docNode attributeForName: @"initialViewController"] stringValue]);             
+      _scenesMap = [[NSMutableDictionary alloc] initWithCapacity: [array count]];
+      _controllerMap = [[NSMutableDictionary alloc] initWithCapacity: [array count]];
+      _documentsMap = [[NSMutableDictionary alloc] initWithCapacity: [array count]];
+      
+      //while ((e = [en nextObject]) != nil)
+      FOR_IN(NSXMLElement*, e, array) 
+        {
+          NSXMLElement *doc = [[NSXMLElement alloc] initWithName: @"document"];
+          NSArray *children = [e children];
+          NSXMLDocument *document = nil;
+          NSString *sceneId = [[e attributeForName: @"sceneID"] stringValue]; 
+          NSString *controllerId = nil;
+
+          // Copy children...
```

Comment:
Actually you are not coming these, just moving.

Reply:
You're right, changed